### PR TITLE
Fix remaining `MS_SHOULD_BE_FINAL` SpotBugs violations

### PR DESCRIPTION
**Pull request by basil**
Fixes the remaining `MS_SHOULD_BE_FINAL` SpotBugs violations. See self-review for details.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@StefanSpieker

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/java/hudson/FilePath.java
Comment (basil) [inline core/src/main/java/hudson/FilePath.java:3313]: I searched for usages in sources (with GitHub) and binaries (with `usage-in-plugins`) and could not find any references in the `jenkinsci` GitHub organization or in plugin binaries, so this is safe to restrict. Since it is restricted, it is safe to mark it final, as no plugins will inline the value during compilation.

File: core/src/main/java/jenkins/model/Jenkins.java
Comment (basil) [inline core/src/main/java/jenkins/model/Jenkins.java:5509]: It seems clear from context that this was intended to be a tunable rather than a constant. There are no references to it in the `jenkinsci` GitHub organization or in any plugins. I've made this a proper tunable using `SystemProperties` and taken the liberty to rename it to "automatic agent launch". Once this PR is merged I'll follow up with another PR to add the tunable to the documentation on `jenkins.io`.

File: core/src/main/java/jenkins/model/item_category/ItemCategory.java
Comment (basil) [inline core/src/main/java/jenkins/model/item_category/ItemCategory.java:51]: Since this is restricted, it is safe to mark it final, as no plugins will inline the value during compilation. I verified that there were no references to this in the `jenkinsci` GitHub organization or in any plugin binaries.

**Review by timja — APPROVED** [review]

**Review by StefanSpieker — APPROVED** [review]

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
